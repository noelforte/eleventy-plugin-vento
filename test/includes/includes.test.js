import { runBuild } from '../get-instance.js';
import { expect, test } from 'vitest';

console.log(import.meta.dirname);

const results = await runBuild(import.meta.dirname);

test('include a page', () => {
	const page = results.find(({ url }) => url === '/basic/');
	expect(page.content).toMatchSnapshot();
});

test('include a page with data passing', async () => {
	const page = results.find(({ url }) => url === '/with-data/');
	await expect(page.content).toMatchSnapshot();
});
