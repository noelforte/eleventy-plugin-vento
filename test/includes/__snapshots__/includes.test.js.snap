// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`include a page 1`] = `
"
<h1>Page with includes</h1>

<section class="include">
	<p>This content is included.</p>
	<p>Passed data for \`extra\` key: ''</p>
</section>

"
`;

exports[`include a page with data passing 1`] = `
"
<h1>Page with includes (data passing)</h1>

<section class="include">
	<p>This content is included.</p>
	<p>Passed data for \`extra\` key: 'super fancy data'</p>
</section>

"
`;
